src/graphql/typeDefs.js

```diff
@@ -23,19 +23,21 @@ module.exports = gql`
   }
 
   type Thread {
+    # id of the message
     id: ID
-    messages: [Comment]
-    # repliesCount: Int
-  }
 
-  type Comment {
-    id: ID
+    # link to root message, if there is one
+    root: Thread
```

Comment:
:fire: I think I would be tempted to make this 
```suggestion
    rootId: String
```

I'm imagining a UI where you load up a "thread" from some random message, and you want to indicate there is stuff "upstream" of this thread, that this is a part of.
I want  a UI like

```js
<div v-if="thread.rootId">
  see root thread <a href=`/thread/${thread.rootId}`>here</a>     // rootId = %A
</div>

( Cherese: Some comment mid-thread )          // id %C
( Mix: this is a reply to Cherese's comment ) // id %D
```

for thread like
```mermaid
flowchart TB

A-->B

subgraph subThread
  C-->D[D: mix reply]
end
B-->C-->E

```

Reply:
I think only `rootId` is needed. If you let people accidentally load the whole `root` Thread in there then it will do two queries aye, and you'll get A,B,C,D,E all loaded in... when you only really want to have C,D, and know that A _exists_ so you can click to it.

DESIGN QUESTION?

I would be inclined to only allow minimal loading and only add more later IF needed